crates/optimism/rpc/src/eth/transaction.rs

```diff
@@ -5,22 +5,30 @@ use alloy_primitives::{Bytes, B256};
 use alloy_rpc_types_eth::TransactionInfo;
 use op_alloy_consensus::{transaction::OpTransactionInfo, OpTransaction};
 use reth_optimism_primitives::DepositReceipt;
-use reth_primitives_traits::SignedTransaction;
+use reth_primitives_traits::{NodePrimitives, SignedTransaction};
 use reth_rpc_eth_api::{
-    helpers::{spec::SignersForRpc, EthTransactions, LoadTransaction},
-    try_into_op_tx_info, FromEthApiError, RpcConvert, RpcNodeCore, TxInfoMapper,
+    helpers::{spec::SignersForRpc, EthTransactions, LoadReceipt, LoadTransaction},
+    try_into_op_tx_info, FromEthApiError, FromEvmError, RpcConvert, RpcNodeCore, RpcReceipt,
+    TxInfoMapper,
 };
 use reth_rpc_eth_types::utils::recover_raw_transaction;
-use reth_storage_api::{errors::ProviderError, ReceiptProvider};
+use reth_storage_api::{errors::ProviderError, ReceiptProvider, TransactionsProvider};
 use reth_transaction_pool::{
     AddedTransactionOutcome, PoolTransaction, TransactionOrigin, TransactionPool,
 };
-use std::fmt::{Debug, Formatter};
+use std::{
+    fmt::{Debug, Formatter},
+    future::Future,
+};
 
 impl<N, Rpc> EthTransactions for OpEthApi<N, Rpc>
 where
-    N: RpcNodeCore,
+    N: RpcNodeCore<
+        Provider: TransactionsProvider<Transaction = <N::Primitives as NodePrimitives>::SignedTx>,
+    >,
+    OpEthApiError: FromEvmError<N::Evm>,
     Rpc: RpcConvert<Primitives = N::Primitives, Error = OpEthApiError>,
+    Self: RpcNodeCore<Primitives = N::Primitives, Pool = N::Pool>,
```

Comment:
@klkvr had to restrict this like this so we can reuse treat `OpEthApi as RpcNodeCore == N`
but perhaps there's a better way?